Add support and tests for SSL on custom domains in CDN

nachakra opened the pull request:
Link to swagger PR - https://github.com/Azure/azure-rest-api-specs/pull/5276

nachakra committed:
Add support and tests for SSL on custom domains in CDN

nachakra committed:
Fix the test failure for profiles

nachakra committed:
Fix the build and refactor the profile tests

Since we use an existing susbscription that is used for other testing too, the tests failed beacuse they assumed it is a clean subscription. I changed that assumption so that they(hopefully) work with any subscription in record mode.

nachakra committed:
Update nuget package version

shahabhijeet reviewed (changes requested):
You are missing Swagger PR information.
Paste a link to the swagger PR.
Also update nuget package version in .csproj, also update version in AssemblyInfo.cs
Generate SDK using generate.ps1, which will essentially generate all the meta data files that are required for .NET SDK

nachakra commented:
> You are missing Swagger PR information.
> Paste a link to the swagger PR.
> Also update nuget package version in .csproj, also update version in AssemblyInfo.cs
> Generate SDK using generate.ps1, which will essentially generate all the meta data files that are required for .NET SDK

nachakra committed:
Update file version

nachakra commented:
@shahabhijeet -
1) I added the swagger link to the description. 
2) I updated the assemblyinfo.cs. When do we need to change the assemblyinfo.cs if it is not a breaking change. Also it is not a new version of API. 
.csproj is included in the PR and the nuget package version is already updated.
3) I did use the generate.cmd to generate the SDK and I do see a metadata file in this PR - /SDKs/_metadata/cdn_resource-manager.txt
Should there be more?

shahabhijeet commented on src/SDKs/Cdn/Cdn.Tests/ScenarioTests/CustomDomainTests.cs:
```diff
@@ -112,17 +112,40 @@ public void CustomDomainCRUDTest()
+                var byocParameters = new KeyVaultCertificateSourceParameters()
+                {
+                    ResourceGroupName = "byoc",
+                    SecretName = "CdnSDKE2EBYOCTest",
```
@nachakra no hardcoding secrets.
Please use TestFramework to inject custom data.
https://github.com/Azure/azure-sdk-for-net/blob/master/Documentation/Using-Azure-TestFramework.md#1-once-you-set-your-connection-string-you-can-add-or-update-keyvalue-settings

nachakra replied:
This is the cert name in Keyvault and not a secret :). That's how the property is called in the API model.

shahabhijeet commented on src/SDKs/Cdn/Management.Cdn/Generated/CdnManagementClient.cs:
```diff
@@ -397,6 +397,8 @@ private void Initialize()
             DeserializationSettings.Converters.Add(new PolymorphicDeserializeJsonConverter<DeliveryRuleAction>("name"));
             SerializationSettings.Converters.Add(new PolymorphicSerializeJsonConverter<DeliveryRuleCondition>("name"));
             DeserializationSettings.Converters.Add(new PolymorphicDeserializeJsonConverter<DeliveryRuleCondition>("name"));
+            SerializationSettings.Converters.Add(new PolymorphicSerializeJsonConverter<CustomDomainHttpsParameters>("certificateSource"));
```
@nachakra did you add this after the generated code, if yes, all these custom settings should be added inside CustomInitialize() that you will be defining in your customization layer.
Create a customization folder and add a new class file with the same exact class name (as all classes partial)
Declare CustomInitialize() in it and add whatever custom code you need.

nachakra replied:
I did not change anything in the generated code. Just refactored the tests.

shahabhijeet commented on src/SDKs/Cdn/Management.Cdn/Microsoft.Azure.Management.Cdn.csproj:
```diff
@@ -5,11 +5,11 @@
 	<PropertyGroup>
 		<Description>Microsoft Azure CDN Management Library</Description>
-		<Version>4.0.2-preview</Version>
+		<Version>4.0.3-preview</Version>
```
you added a new functionality.
Update minor version, so make
nuget version = 4.1.2-preview
assemblyfileversion = 4.1.2

nachakra replied:
Sure.

nachakra committed:
Update the moinor version since new functionality was added

nachakra commented:
@shahabhijeet - can you take a look again at the PR please?

dsgouda reviewed (approved):
LGTM

shahabhijeet reviewed (changes requested):
If you are planning to release this SDK, please update version info in .csproj and AssemblyInfo.cs
Also add release notes.

nachakra commented:
@shahabhijeet  - I updated both in the last commit or the commits before. Not sure what I am still missing.
